### ISSUE TITLE
Excessive memory usage after upgrading to v2 from v1 and Go running garbage collector every two minutes

### ISSUE THREAD

**Issue by xPaw** [influxdata/influxdb#22275]
__Actual behavior:__

I was using InfluxDB 1.8 (and lower versions) for a long time, and it consumed around ~1gb memory consistently, and I did not see any cpu spiking. After upgrading to v2, it uses 7gb of real memory and 12gb virtual. The data folder is just 2.5gb, I find this memory usage exessive.

The bigger problem is that every two minutes the Go garbage collector runs (can repro with /pprof/heap?gc=1), and it spikes multiple cores to 100% usage for a second.

I'm certainly not alone with this, I can see various posts on the forum about the memory usage, for example this one:
https://community.influxdata.com/t/influx-freezes-vm-maxing-out-memory/17960/8

However I don't see a concrete issue about it. What changed in v2 so drastically that it needs to consume so much more memory?

![image](https://user-images.githubusercontent.com/613331/130362139-70cdaf48-c123-4c5a-bd43-2c9ee59a2118.png)

__Environment info:__

* System info: Linux 4.19.0-14-amd64 x86_64
* InfluxDB version: InfluxDB 2.0.8 (git: e91d41810f) build_date: 2021-08-13T18:22:30Z
* Other relevant environment details: Debian 10

__Config:__
Copy any non-default config values here or attach the full config as a gist or file.

```
log-level = "warn"
reporting-disabled = true
```

__Performance:__

[profiles.tar.gz](https://github.com/influxdata/influxdb/files/7027679/profiles.tar.gz)

EDIT:
I looked at heap pprof and it seems to be most memory used as in #11830

```
      flat  flat%   sum%        cum   cum%
    2.13GB 47.08% 47.08%        4GB 88.35%  github.com/influxdata/influxdb/v2/tsdb/index/tsi1.(*LogFile).execSeriesEntry
    1.12GB 24.84% 71.92%     1.12GB 24.84%  github.com/influxdata/influxdb/v2/tsdb/index/tsi1.(*logTagValue).addSeriesID
    0.68GB 14.92% 86.84%     0.68GB 14.92%  github.com/influxdata/influxdb/v2/tsdb/index/tsi1.(*logTagKey).createTagValueIfNotExists (inline)
    0.21GB  4.57% 91.41%     0.21GB  4.57%  github.com/influxdata/influxdb/v2/pkg/estimator/hll.(*Plus).toNormal
    0.06GB  1.27% 92.68%     0.06GB  1.27%  bytes.makeSlice
    0.06GB  1.27% 93.95%     0.06GB  1.27%  github.com/influxdata/influxdb/v2/tsdb/index/tsi1.(*logTagValue).seriesIDSet
    0.05GB  1.10% 95.05%     0.06GB  1.37%  github.com/RoaringBitmap/roaring.(*arrayContainer).iaddReturnMinimized
    0.03GB  0.76% 95.81%     0.03GB  0.76%  github.com/influxdata/influxdb/v2/pkg/rhh.(*HashMap).alloc (inline)
    0.03GB   0.6% 96.41%     0.03GB   0.6%  bufio.NewWriterSize
    0.01GB  0.32% 96.73%     0.06GB  1.39%  github.com/influxdata/influxdb/v2/tsdb.(*SeriesIndex).execEntry
```

**Comment by danxmoran** [comment]
@xPaw were you using the `tsi1` index type when running 1.8? Or the `inmem` type?

**Comment by xPaw** [comment]
Inmem mostly, I did switch to TSI just before upgrading to v2.

**Comment by xPaw** [comment]
After upgarding to 2.0.9 memory usage went down from 7gb to 1.2gb (it still has 6.1gb virtual memory). I will monitor how it will behave after a couple of days.

**Comment by lesam** [comment]
This was likely fixed in 2.0.9 by https://github.com/influxdata/influxdb/pull/22520
